tests/storage/guestfs.go

```diff
@@ -0,0 +1,167 @@
+		}()
+		// Waiting until the libguestfs pod is running
+		Eventually(func() corev1.PodPhase {
+			pod, _ := virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Get(context.Background(), podName, metav1.GetOptions{})
+			return pod.Status.Phase
+		}, 90*time.Second, 2*time.Second).Should(Equal(k8sv1.PodRunning))
+		// Verify that the appliance has been extracted before running any tests
+		Eventually(func() bool {
+			output, _, err := execCommandLibguestfsPod(podName, []string{"ls", "/usr/local/lib/guestfs/appliance"})
+			Expect(err).ToNot(HaveOccurred())
+			if strings.Contains(output, "kernel") {
+				return true
+			}
+			return false
+		}, 30*time.Second, 2*time.Second).Should(BeTrue())
+
+	}
+
+	createGuestfsWithPVC := func(pvc *corev1.PersistentVolumeClaim) {
+		var err error
+		_, err = virtClient.CoreV1().PersistentVolumeClaims(tests.NamespaceTestDefault).Create(context.Background(), pvc, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		runGuestfsOnPVC(pvc.ObjectMeta.Name)
+
+	}
+
+	Context("Run libguestfs on PVCs", func() {
+		BeforeEach(func() {
+			var err error
+			virtClient, err = kubecli.GetKubevirtClient()
+			Expect(err).ToNot(HaveOccurred())
+
+		}, 120)
+
+		// libguestfs-test-tool verifies the setup to run libguestfs-tools
+		It("Should successfully run libguestfs-test-tool", func() {
+			f := createFakeAttacher()
+			defer f.closeChannel()
+			pvcClaim := "pvc-verify"
+			pvc := createPVCFilesystem(pvcClaim)
+			createGuestfsWithPVC(pvc)
+			output, _, err := execCommandLibguestfsPod("libguestfs-tools-"+pvcClaim, []string{"libguestfs-test-tool"})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output).To(ContainSubstring("===== TEST FINISHED OK ====="))
+
+		})
+
+		It("Should successfully run guestfish on a filesystem-based PVC", func() {
+			f := createFakeAttacher()
+			defer f.closeChannel()
+			pvcClaim := "pvc-fs"
+			podName := "libguestfs-tools-" + pvcClaim
+			pvc := createPVCFilesystem(pvcClaim)
+			createGuestfsWithPVC(pvc)
+			stdout, stderr, err := execCommandLibguestfsPod(podName, []string{"qemu-img", "create", "/disks/disk.img", "500M"})
+			Expect(stderr).To(Equal(""))
+			Expect(stdout).To(ContainSubstring("Formatting"))
+			Expect(err).ToNot(HaveOccurred())
+			stdout, stderr, err = execCommandLibguestfsPod(podName, []string{"guestfish", "-a", "/disks/disk.img", "run"})
+			Expect(stderr).To(Equal(""))
+			Expect(stdout).To(Equal(""))
+			Expect(err).ToNot(HaveOccurred())
+
+		})
+
+		It("Should fail run guestfs command on a PVC in use", func() {
+			f := createFakeAttacher()
+			defer f.closeChannel()
+			pvcClaim := "pvc-fail-to-run-twice"
+			pvc := createPVCFilesystem(pvcClaim)
+			createGuestfsWithPVC(pvc)
+			guestfsCmd := tests.NewVirtctlCommand("guestfs",
+				"--pvc-namespace", tests.NamespaceTestDefault)
+			Expect(guestfsCmd.Execute()).To(HaveOccurred())
+		})
+
+		It("Should successfully run guestfish on a block-based PVC", func() {
```

Comment:
Request adding Polarion Test case id as follows:

Replace "Should successfully run guestfish on a block-based PVC" with:
              "[posneg:positive][test_id:6479]Should successfully run guestfs on a block-based PVC"